demos/multi_channel/object_detection_demo_yolov3/main.cpp

```diff
@@ -0,0 +1,617 @@
+    std::cout << "    -nc                          " << num_cameras << std::endl;
+    std::cout << "    -bs                          " << batch_size << std::endl;
+    std::cout << "    -nireq                       " << num_infer_requests << std::endl;
+    std::cout << "    -n_iqs                       " << input_queue_size << std::endl;
+    std::cout << "    -fps_sp                      " << fps_sampling_period << std::endl;
+    std::cout << "    -n_sp                        " << num_sampling_periods << std::endl;
+    std::cout << "    -pc                          " << performance_counter_message << std::endl;
+    std::cout << "    -t                           " << thresh_output_message << std::endl;
+    std::cout << "    -no_show                     " << no_show_processed_video << std::endl;
+    std::cout << "    -show_stats                  " << show_statistics << std::endl;
+    std::cout << "    -duplicate_num               " << duplication_channel_number << std::endl;
+    std::cout << "    -real_input_fps              " << real_input_fps << std::endl;
+    std::cout << "    -i                           " << input_video << std::endl;
+}
+
+bool ParseAndCheckCommandLine(int argc, char *argv[]) {
+    // ---------------------------Parsing and validation of input args--------------------------------------
+    gflags::ParseCommandLineNonHelpFlags(&argc, &argv, true);
+    if (FLAGS_h) {
+        showUsage();
+        showAvailableDevices();
+        return false;
+    }
+    slog::info << "Parsing input parameters" << slog::endl;
+
+    if (FLAGS_m.empty()) {
+        throw std::logic_error("Parameter -m is not set");
+    }
+    if (FLAGS_nc == 0 && FLAGS_i.empty()) {
+        throw std::logic_error("Please specify at least one video source(web cam or video file)");
+    }
+    slog::info << "\tDetection model:           " << FLAGS_m << slog::endl;
+    slog::info << "\tDetection threshold:       " << FLAGS_t << slog::endl;
+    slog::info << "\tUtilizing device:          " << FLAGS_d << slog::endl;
+    if (!FLAGS_l.empty()) {
+        slog::info << "\tCPU extension library:     " << FLAGS_l << slog::endl;
+    }
+    if (!FLAGS_c.empty()) {
+        slog::info << "\tCLDNN custom kernels map:  " << FLAGS_c << slog::endl;
+    }
+    slog::info << "\tBatch size:                " << FLAGS_bs << slog::endl;
+    slog::info << "\tNumber of infer requests:  " << FLAGS_nireq << slog::endl;
+    slog::info << "\tNumber of input web cams:  "  << FLAGS_nc << slog::endl;
+
+    return true;
+}
+
+static int EntryIndex(int side, int lcoords, int lclasses, int location, int entry) {
+    int n = location / (side * side);
+    int loc = location % (side * side);
+    return n * side * side * (lcoords + lclasses + 1) + entry * side * side + loc;
+}
+
+struct YoloParams {
+    int num;
+    int classes;
+    int coords;
+
+    std::vector<float> anchors;
+
+    YoloParams(int num, int classes, int coords, std::vector<float>anchors) {
```

Comment:
I doubt you need this constructor. It is enough to rely on [struct initialization](https://en.cppreference.com/w/c/language/struct_initialization). It will even handle support for move semantics (in case if it would have been used), while current approach forces `anchors` coping. Feel free to ignore that nitpicking because two other demos don't use that feature.